python/dask_cudf/dask_cudf/backends.py

```diff
@@ -115,21 +115,14 @@ def concat_cudf(
 try:
 
     from dask.dataframe.utils import group_split_dispatch, hash_object_dispatch
-    from cudf.core.column import column, CategoricalColumn, StringColumn
+    from cudf.core.column import column
 
-    def _handle_string(s):
-        if isinstance(s._column, StringColumn):
-            s = s._hash()
-        return s
-
-    def safe_hash(df):
-        frame = df.copy(deep=False)
+    def safe_hash(frame):
+        index = frame.index
         if isinstance(frame, cudf.DataFrame):
-            for col in frame.columns:
-                frame[col] = _handle_string(frame[col])
-            return frame.hash_columns()
+            return cudf.Series(frame.hash_columns(), index=index)
         else:
-            return _handle_string(frame)
+            return cudf.Series(frame.hash_values(), index=index)
```

Comment:
This function likely won't exist for a MultiIndex, do we need to handle that here?

Reply:
A multiIndex should be routed to `hash_object_cudf_index` and will be converted to a frame before hashing. `cudf.MultiIndex` is an instance of `cudf.Index`, correct?

Reply:
Yes, `cudf.MultiIndex` inherits from `cudf.Index`